src/gwt/src/org/rstudio/studio/client/rmarkdown/events/RmdChunkOutputFinishedEvent.java

```diff
@@ -52,6 +55,18 @@ public final native int getType() /*-{
       public final native int getScope() /*-{
          return this.scope;
       }-*/;
+      
+      public final ArrayList<String> getHtmlCallback() {
+         JsArray arr = getNativeHtmlCallbacks();
+         ArrayList<String> results = new ArrayList<>();
+         for (int i = 0; i < arr.length(); i++)
+            results.add(arr.get(i).toString());
+         return results;
+      }
+
+      private native final JsArray getNativeHtmlCallbacks() /*-{
+         return this["html_callback"][0] || this["html_callback"]["html"] || [];
```

Comment:
This feels a bit awkward: can we normalize the return value beforehand so that we don't have to "guess" where the HTML array lives?

Reply:
Academy requested the callback function return a list in the original specs so I'm following up with them to see if it can return a string since that would make things simple.

Reply:
The callback function returns a list with 'html' in it now in case we want to add other variables in the future. I moved all the management of that object to the back end code.